### ISSUE TITLE
[BUG] Unable to turn off build progress and ansi symbols

### ISSUE THREAD

**Issue by burguy** [docker/compose#10825]
### Description

None of command line options that might quiet the build process works as expected.
I see the progress of image pulling, docker commands from the docker file, and ansi symbols are always present.

### Steps To Reproduce

`docker compose --ansi never --progress quiet -f .\docker-compose.build.yml build --quiet`

The command output looks as following:
```txt
Sending build context to Docker daemon     331B
Sending build context to Docker daemon     236B
Step 1/6 : ARG BASE_IMAGE
Step 2/6 : ARG INTEGRATION_RESOURCES_IMAGE
Sending build context to Docker daemon     330B
Step 3/6 : FROM ${INTEGRATION_RESOURCES_IMAGE} AS resources
 ---> e1619dc6b2e0
Step 4/6 : FROM ${BASE_IMAGE}
Step 1/7 : ARG BASE_IMAGE
Step 2/7 : ARG INTEGRATION_RESOURCES_IMAGE
Step 3/7 : FROM ${INTEGRATION_RESOURCES_IMAGE} AS resources
 ---> e1619dc6b2e0
Step 4/7 : FROM ${BASE_IMAGE}
Step 1/7 : ARG BASE_IMAGE
Step 2/7 : ARG INTEGRATION_RESOURCES_IMAGE
Step 3/7 : FROM ${INTEGRATION_RESOURCES_IMAGE} AS resources
 ---> e1619dc6b2e0
Step 4/7 : FROM ${BASE_IMAGE}
<tag-hidden>: Pulling from <image-hidden>
<tag-hidden>: Pulling from <image-hidden>
<tag-hidden>: Pulling from <image-hidden>
b1471cc22404: Already exists
b1471cc22404: Already exists
7b80e910ae7d: Pulling fs layer
3c6669ab9a5e: Pulling fs layer
c1f42d4dfd33: Pulling fs layer
44ccfa611bed: Pulling fs layer
8b7c4a5e6993: Waiting
957c06eb0610: Pulling fs layer
271199e4a776: Waiting
de8f6a2f938d: Waiting
27b95426b3aa: Waiting
0cdf5c0cf6bd: Waiting
859d0fc68013: Waiting
3cf6f62f61eb: Waiting
b73b6d9d4921: Pulling fs layer
b73b6d9d4921: Pulling fs layer
101d4064bd12: Waiting
83b20833b8dd: Waiting
3cf6f62f61eb: Waiting
779e17f9bea6: Waiting
35c89b689551: Waiting
e3af75e99335: Waiting
779e17f9bea6: Waiting
748790eb0436: Waiting
27b95426b3aa: Waiting
b54efb111144: Waiting
7b80e910ae7d: Pull complete
3c6669ab9a5e: Pull complete
c1f42d4dfd33: Extracting [==================================================>]  1.409kB/1.409kB
44ccfa611bed: Download complete
231c9be85a4a: Download complete
5e967e1c5835: Download complete
```

### Compose Version

```Text
Docker Compose version v2.19.1
```

### Docker Environment

```Text
Client:
 Version:    24.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  compose: Docker Compose (Docker Inc.)
    Version:  v2.19.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 62
 Server Version: 24.0.2
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: process
 Kernel Version: 10.0 17763 (17763.1.amd64fre.rs5_release.180914-1434)
 Operating System: Microsoft Windows Server Version 1809 (OS Build 17763.4645)
 OSType: windows
 Architecture: x86_64
 CPUs: 8
 Total Memory: 32GiB
 Name: vm
 ID: 11111111-2222-3333-4444-555555555555
 Docker Root Dir: C:\ProgramData\docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

### Anything else?

Docker Compose options: https://docs.docker.com/engine/reference/commandline/compose/
See options `--ansi` and `--progress`

Docker Compose Build options: https://docs.docker.com/engine/reference/commandline/compose_build/
See option `--quiet`

**Comment by liudonghua123** [comment]
It seems `--progress quiet` is not available in the new version of `docker-compose`.

```
@liudonghua123 ➜ /workspaces/php-wasm (master) $ docker-compose -v
docker-compose version 1.29.2, build 5becea4c
@liudonghua123 ➜ /workspaces/php-wasm (master) $ docker-compose build --help
Build or rebuild services.

Services are built once and then tagged as `project_service`,
e.g. `composetest_db`. If you change a service's `Dockerfile` or the
contents of its build directory, you can run `docker-compose build` to rebuild it.

Usage: build [options] [--build-arg key=val...] [--] [SERVICE...]

Options:
    --build-arg key=val     Set build-time variables for services.
    --compress              Compress the build context using gzip.
    --force-rm              Always remove intermediate containers.
    -m, --memory MEM        Set memory limit for the build container.
    --no-cache              Do not use cache when building the image.
    --no-rm                 Do not remove intermediate containers after a successful build.
    --parallel              Build images in parallel.
    --progress string       Set type of progress output (auto, plain, tty).
    --pull                  Always attempt to pull a newer version of the image.
    -q, --quiet             Don't print anything to STDOUT
@liudonghua123 ➜ /workspaces/php-wasm (master) $
```


Option | Short | Default | Description
-- | -- | -- | --
--progress |   | auto | Set type of ui output (auto, tty, plain, quiet)

See also https://docs.docker.com/engine/reference/commandline/compose_build/#options.

**Comment by liudonghua123** [comment]
> It seems `--progress quiet` is not available in the new version of `docker-compose`.
> 
> ```
> @liudonghua123 ➜ /workspaces/php-wasm (master) $ docker-compose -v
> docker-compose version 1.29.2, build 5becea4c
> @liudonghua123 ➜ /workspaces/php-wasm (master) $ docker-compose build --help
> Build or rebuild services.
> 
> Services are built once and then tagged as `project_service`,
> e.g. `composetest_db`. If you change a service's `Dockerfile` or the
> contents of its build directory, you can run `docker-compose build` to rebuild it.
> 
> Usage: build [options] [--build-arg key=val...] [--] [SERVICE...]
> 
> Options:
>     --build-arg key=val     Set build-time variables for services.
>     --compress              Compress the build context using gzip.
>     --force-rm              Always remove intermediate containers.
>     -m, --memory MEM        Set memory limit for the build container.
>     --no-cache              Do not use cache when building the image.
>     --no-rm                 Do not remove intermediate containers after a successful build.
>     --parallel              Build images in parallel.
>     --progress string       Set type of progress output (auto, plain, tty).
>     --pull                  Always attempt to pull a newer version of the image.
>     -q, --quiet             Don't print anything to STDOUT
> @liudonghua123 ➜ /workspaces/php-wasm (master) $
> ```
> 
> Option	Short	Default	Description
> --progress	 	auto	Set type of ui output (auto, tty, plain, quiet)
> See also https://docs.docker.com/engine/reference/commandline/compose_build/#options.

I didn't notice I used the old version of `docker-compose`. 🥲

**Comment by glours** [comment]
Hello @burguy @liudonghua123 
Can you try with one of the latest version of Compose please? 
I wasn't able to reproduce even with the version `v2.19.1`, if you still have the issue, can you share more details about your Compose and Docker file please? Or try to reproduce with [one of the `awesome-compose` samples](https://github.com/docker/awesome-compose/)?

**Comment by glours** [comment]
As mentioned in the [contributing guide](https://github.com/docker/compose/blob/main/CONTRIBUTING.md), without any feedback to our last comment for a while  I'm closing the issue

**Comment by ShadowLNC** [comment]
@glours I am also seeing issues with `--ansi` not doing anything - the output does not appear to change whether I specify `never`, `always`, or `0`.

I am running `docker compose --ansi=<value> --env-file ... --env-file ... up --wait <service_name>`.

My `pull_policy` is set to `always`, and no containers are being built. I tried briefly setting `pull_policy` to `never`, omitting the `--env-file` arguments, and changing `--wait` to `-d`, to no avail.

macOS Sonoma 14.2.1 - zsh / VSCode terminal
Docker Compose version v2.23.3-desktop.2
as part of Docker Desktop 4.26.1

I am trying to turn off formatting in the hope that it will resolve #9962 #8800 and the still-open #10256 #11254 but for now I'm stuck with this behaviour 😞 

Bonus: I noticed that `docker compose up --help` lists `--no-color` as a flag, but when I use it, it tells me it's unrecognised.
